### PR TITLE
Split image into dask and jhub, build both and tag accordingly

### PR DESCRIPTION
**Pull request by tom-butler**
It's a bit heavy handed but should work

### CODE REVIEW

File: docker/adaptive.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        keep_running = False
+
+    signal.signal(signal.SIGTERM, sigterm_handler)
+    min_pods = dask.config.get('kubernetes.count.min')
+    if min_pods is None:
+        min_pods = 0
+    with KubeCluster() as cluster:
+        cluster.adapt(minimum=min_pods)
+        while keep_running:
+            sleep(0)
```
Comment (woodcockr) [inline docker/adaptive.py:27]: Busy wait?

File: .travis.yml [hunk 2]
```diff
@@ -3,32 +3,30 @@ sudo: required  # Use the new travis docker based infrastructure
...
-  slack:
-    secure: cNYu6iz861hlq8vmkIXGbsnVTzfx0P2hWEFJp9baFh40GuTtuU8qfeQvBtWguRMsCCbvE2Pwr1UZQQxDWUK2hLGqi+nT9kmItpRoGH69fvyXLWrEfU3HERanpBYine949fEksSlMCv3BuqZqCf9rtfbZp743jErQVctJkD+Pf1rC/Qt0eK9x9MykPE+lIupm6j1t9aFJuAF6lfcesGNYuYl0W4Vjcc+iLrZG/7s/ixQWlgxGTHVkVHHed2rsv6JHbm+9WPchPXvD12o5tdmFHqnzog30rA10V2eM03tYcTaExPUe+Z1RLv9tK/A//mTO35wg7laB0AqxFaCS+s4R3wdYpq/IvFIyiJp96ww+u9S1j9jnZZ/TxuB3ilGVUoKRr92AVxL5ghlhL0+a4QJEwOJ1BTytlc1eh0fXN5xKtzqwEPOQqP9jNhTPZP44attEc2knpFSTaIoin6LTg6LptlQFLbqhtNFc+kFK7OZMMNZwVI+oxyqAEaE1hgMRYKukNyUs6YmGzCX935Pe84Kq/Oo63+d05HN0qxHfXWxKCkdX9pagJVLMF4jI78abRPPjtmKC7L+G1//ZKdSB6sHTpWT7vcUCCPUpIXLTGsKU2EZ0r9to+5nQ/PjFJDZta5VfLDkcf9bK+r5nWrQafzHl8x0rpgZpOtFUXPCAutEc7Y0=
+language: python
+python:
+# We don't use the travis python, but we will use it for the build matrix
+  - "3.6"
+jobs:
+  include:
+    - stage: build
+      if: (branch = master OR (tag = branch)) AND type = push
```
Comment (Kirill888) [inline .travis.yml:23]: what was the need to revert `.travis.yml` changes?

Comment (Kirill888) [inline .travis.yml:23]: I want my travis to build image no matter what branch, not only after PR is accepted

File: docker/adaptive.py (hunk 1 above)
Comment (tom-butler) [reply docker/adaptive.py:27]: we're just merging https://github.com/opendatacube/datacube-dask and this repo so they are built using the same libraries, while I admit this is an issue, I'm not going to fix it in this PR

**Review by Kirill888 — CHANGES_REQUESTED** [review]
travis build must happen on all branches, push to docker hub needs to happen on specific branches.

Current PR doesn't even build suggested changes, how do we know it works?

**Comment by Kirill888** [comment]
I'm happy with docker file changes, but not with chages to `.travis.yml`. having stages is costly because build docker images do not survive between stages, that's why I had `.travis.yml` changed back then. I understand that  you just used some common template you have but it doesn't work very well in this scenario. And "only build after PR is merged" is not a strategy that will work in this case, it's a complicated docker file with all the lab extensions and all.

Can you please add the extra steps needed like that security scan and all to the `deploy.sh` and leave `.travis.yml` mostly unchanged.

**Pull request closed without merging**